[CDAP-6745] [CDAP-6174] Integrates with tracker

PGuimarais opened the pull request:
- Integrates LogViewer with tracker
- Provides full-screen solution for navigator
- Changes artifact version

Fixes Bugs:
- https://issues.cask.co/browse/CDAP-6745
- https://issues.cask.co/browse/CDAP-5339
- https://issues.cask.co/browse/CDAP-6840

albertshau committed:
CDAP-6372 add spark streaming app

Added apis for new StreamingSource and Windower plugin
types for the data streams application.

Also doing some more refactoring on the spark program so that both
the data streams and data pipeline apps can share almost all the
same logic. Introduced the SparkCollection and SparkPairCollection
interfaces which the main driver performs operations on.
The data pipeline app implements them with RDDs while the data
streams app implements it with DStreams.

PGuimarais committed:
Adds download functionality

PGuimarais committed:
Enables different naming conventions for log file download

PGuimarais committed:
Fixes download name

astewes committed:
fixes positioning issue with time/date picker

PGuimarais committed:
Changes timeline labels from 12hr to 24 hr format

PGuimarais committed:
Integrates with tracker

PGuimarais committed:
Integrates with tracker

astewes committed:
fixes tag text and rendering issues

PGuimarais committed:
Uses url with controlleras syntax

PGuimarais committed:
Fixes full-screen mode

removes padding from logviewer in tracker

Places application status in headerbar

Updates full-screen to include program status in headerbar

PGuimarais committed:
Fixes full-screen mode

removes padding from logviewer in tracker

Places application status in headerbar

Updates full-screen to include program status in headerbar

elfenheart commented:
![screen shot 2016-07-27 at 11 22 59 am](https://cloud.githubusercontent.com/assets/4398643/17187045/85f6985e-53ec-11e6-9aea-800d0b7ba5ab.png)

there are paddings on the side of the log viewer

astewes committed:
adds ng-pluralize

elfenheart commented:
👍  LGTM

astewes committed:
adds beta text to tracker meter

PGuimarais committed:
Fixes broken download button

Fixes bug where stack traces collapse themselves

Removes unnecessary function

removes unnecessary scope variable

PGuimarais committed:
Cleanup

Cleanup

Adds status update to getStatus function

Refactors setting status into helper function

Cleanup

PGuimarais committed:
Cleanup

Cleanup

Adds status update to getStatus function

Refactors setting status into helper function

Cleanup

astewes committed:
adds conditional display of beta text

lumig242 committed:
CDAP-6794 Support cross namespace write for map reduce program through output class

elfenheart committed:
add unauthorized page if user do not have access to any namespaces

albertshau committed:
CDAP-6372 add java opts config option

Making the extra java opts a configuration setting instead of
hardcoding it. Also adding jira references to records in and out.

Commit:
(CDAP-6534) Added methods in Impersonator and NamespacedLocationFactory to accept a NamespaceMeta instead of querying it from the MDS. This is currently used while creating a namespace.

gsps1 committed:
CDAP-6666 impersonation fix StreamFileJanitor cleanup

elfenheart committed:
fix duplicate dependencies

elfenheart commented on cdap-ui/app/directives/log-viewer/log-viewer.js:
```diff
@@ -213,6 +237,8 @@ function LogViewerController ($scope, LogViewerStore, myLogsApi, LOGVIEWERSTORE_
       }).$promise.then(
         (statusRes) => {
+          this.programStatus = statusRes.status;
+
```
don't you want to update the statusType here as well?

elfenheart committed:
force update namespace on login

Commit:
(CDAP-6645) Make authorization cache resilient to failures while fetching authorization policies.

Commit:
Remove and replace instances of "cdap-etl-batch" with "cdap-data-pipeline", and other changes

elfenheart committed:
switch to ng-bind for error, and remove backend down event from namespace service

Commit:
Remove duplicate list.

Commit:
Re-arrange order to be alphabetical

elfenheart committed:
opens up management and modify no namespace message

elfenheart committed:
update message for unauthorized namespace

elfenheart committed:
Merge pull request #6250 from caskdata/feature/ui-security-fixes

[CDAP-6772] [CDAP-6770] Handle Unauthorized to any Namespace

Commit:
Merge pull request #6257 from caskdata/docs/r3.5_remove_duplicate_list

Remove duplicate list in Java Client documentation

sagarkapare committed:
CDAP-6815 Using the correct name of the TrackerService.

PGuimarais committed:
Fixes type bug

PGuimarais committed:
Fixes type bug

Commit:
Fixes per comments.

bdmogal committed:
Merge pull request #6256 from caskdata/fix/auth-service-exception

(CDAP-6645) Make authorization cache resilient to failures while fetc…

albertshau committed:
Merge pull request #6186 from caskdata/feature/CDAP-6372-data-streams-app

CDAP-6372 add spark streaming app

astewes committed:
Merge pull request #6225 from caskdata/feature/ui-styling-tracker-meter-beta

adds beta text to tracker meter

astewes committed:
Merge pull request #6219 from caskdata/feature/ui-styling-fix-tracker-tags

fixes tag text and rendering issues

astewes committed:
Merge pull request #6203 from caskdata/feature/ui-fix-stream-explor-time-date-picker

fixes positioning issue with time/date picker

PGuimarais committed:
removes whitespace

bdmogal committed:
Merge pull request #6254 from caskdata/fix/namespace-lookup

(CDAP-6534) Added methods in Impersonator and NamespacedLocationFacto…

Commit:
Merge pull request #6252 from caskdata/docs/r3.5_remove_cdap-etl-batch

Replace cdap-etl-batch

gsps1 committed:
Merge pull request #6245 from caskdata/fix/stream-janitor-impersonation-fix

CDAP-6666 impersonation fix StreamFileJanitor cleanup

PGuimarais committed:
adds program type to download file name

PGuimarais committed:
Merge pull request #6202 from caskdata/feature/ui-log-viewer-downloadbtn

[6174] Feature/ui log viewer downloadbtn

lumig242 committed:
Merge pull request #6239 from caskdata/feature/cross-namespace-output-mapreduce

CDAP-6794 Support cross namespace write for map reduce program through Output class

Nishith committed:
Use sendString instead of sendJson so that the data does not have quotes arround it. (#6268)

(CDAP-6702) Use sendString instead of sendJson so that the data does not have quotes around them.

Closes https://issues.cask.co/browse/CDAP-6702

sagarkapare committed:
Merge pull request #6259 from caskdata/fix/tracker_service_name

CDAP-6815 Using the correct name of the TrackerService.

PGuimarais committed:
Adds missing comma

PGuimarais committed:
Resolves conflict
